schema.sql

```diff
@@ -1,11 +1,31 @@
 /* Database schema to keep the structure of entire database. */
+DROP TABLE IF EXISTS animals;
+DROP TABLE IF EXISTS owners;
+DROP TABLE IF EXISTS species;
+
+CREATE TABLE species (
+    id int GENERATED ALWAYS AS IDENTITY(MINVALUE 0 START WITH 0 CACHE 20) NOT NULL,
+    PRIMARY KEY (id),
+    name char(250)
+);
+
+CREATE TABLE owners (
+    id int GENERATED ALWAYS AS IDENTITY(MINVALUE 0 START WITH 0 CACHE 20) NOT NULL,
+    PRIMARY KEY (id),
+    full_name char(250),
+    age int
+);
 
 CREATE TABLE animals (
-    id int NOT NULL IDENTITY(1,1),
+    id int GENERATED ALWAYS AS IDENTITY(MINVALUE 0 START WITH 0 CACHE 20) NOT NULL,
     PRIMARY KEY (id),
     name char(250),
     date_of_birth date,
     escape_attempts integer,
     neutered boolean,
-    weight_kg decimal
+    weight_kg decimal,
+    species_id int,
+    owner_id int,
+    CONSTRAINT kf_species FOREIGN KEY(species_id) REFERENCES species(id) ON DELETE CASCADE,
+    CONSTRAINT kf_owners FOREIGN KEY(owner_id) REFERENCES owners(id) ON DELETE CASCADE
 );
```

Comment:
**Optional Change**
- [ ] You don't need to DROP the tables and create new one. You can use `ALTER` command to UPDATE the columns and REFERENCES in your database. Otherwise you could have a lot of trouble with deleting data from production.

Reply:
Thanks, I will apply this 😄